server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -1094,11 +1095,22 @@ export class AgenticChatController implements ChatHandlers {
         tabId: string,
         metric: Metric<CombinedConversationEvent>
     ): ChatResult | ResponseError<ChatResult> {
-        if (isAwsError(err) || (isObject(err) && 'statusCode' in err && typeof err.statusCode === 'number')) {
-            metric.setDimension('cwsprChatRepsonseCode', err.statusCode ?? 400)
-            this.#telemetryController.emitMessageResponseError(tabId, metric.metric, err.requestId, err.message)
+        let errorMessage: string
+        let requestID: string | undefined
+
+        if (err instanceof CodeWhispererStreamingServiceException) {
+            errorMessage = err.message
+            requestID = err.$metadata.requestId
+        } else {
+            errorMessage = 'Not a CodeWhispererStreamingServiceException.'
+            if (err instanceof Error || err?.message) {
+                errorMessage += ` Error is: ${err.message}`
+            }
         }
 
+        metric.setDimension('cwsprChatResponseCode', getHttpStatusCode(err) ?? 0)
```

Comment:
Isn't this now assuming that every single error that comes through here is a backend error? Why did we remove the isAWSError check?